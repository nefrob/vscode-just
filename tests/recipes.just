# Recipe attributes

[no-cd]
[private]
foo:
    echo "foo"

[no-cd, private]
foo:
    echo "foo"

[confirm("Are you sure you want to delete everything?")]
delete-everything:
  rm -rf *

# Recipes

default: (build "main")

build target:
  @echo 'Building {{target}}…'
  cd {{target}} && make

foo:
  -cat foo
  echo 'Done!'

build target:
  @echo "Building {{target}}…"

push target: (build target)
  @echo 'Pushing {{target}}…'

# Extra

test: lint && (build ("a" + ("b" / "c"))) && test
  @echo "test"

default := 'all'

test target tests=default:
  @echo 'Testing {{target}}:{{tests}}…'
  ./test --tests {{tests}} {{target}}

test triple=(arch + "-unknown-unknown") input=(arch / "input.dat"):
  ./test {{triple}}

# Extra case
test nested=("a" + ("b" + "c")):
  ./test {{nested}}
  echo (("a") + ("b" + "c"))

commit MESSAGE *FLAGS:
  git commit {{FLAGS}} -m "{{MESSAGE}}"

test +FLAGS='-q':
  cargo test {{FLAGS}}

foo $bar:
  echo $bar

a:
  echo 'A!'

b: a && c d
  echo 'B!'

c:
  echo 'C!'

d:
  echo 'D!'

@quiet:
  echo hello
  echo goodbye
  @# all done!


# Imports

import 'foo/bar.just'

mod bar
