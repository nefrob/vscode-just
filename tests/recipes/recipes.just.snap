># Recipes
#^^^^^^^^^ source.just comment.line.number-sign.just
>
>default: (build "main")
#^^^^^^^ source.just entity.name.function.just
#       ^ source.just keyword.operator.recipe.end.just
#        ^ source.just
#         ^ source.just
#          ^^^^^ source.just entity.name.function.just
#               ^ source.just
#                ^ source.just string.quoted.double.just string.quoted.double.just
#                 ^^^^ source.just string.quoted.double.just
#                     ^ source.just string.quoted.double.just
#                      ^ source.just
>
>build target:
#^^^^^ source.just entity.name.function.just
#     ^ source.just
#      ^^^^^^ source.just variable.parameter.recipe.just
#            ^ source.just keyword.operator.recipe.end.just
>  @echo 'Building {{target}}…'
#^^ source.just
#  ^ source.just keyword.operator.quiet.just
#   ^^^^^ source.just
#        ^ source.just string.quoted.single.just string.quoted.single.just
#         ^^^^^^^^^ source.just string.quoted.single.just
#                  ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                    ^^^^^^ source.just string.quoted.single.just string.interpolated.escaping.just
#                          ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                            ^ source.just string.quoted.single.just
#                             ^ source.just string.quoted.single.just
>  cd {{target}} && make
#^^^^^ source.just
#     ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#       ^^^^^^ source.just string.interpolated.escaping.just
#             ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#               ^ source.just
#                ^^ source.just keyword.operator.and.just
#                  ^^^^^^ source.just
>
>foo:
#^^^ source.just entity.name.function.just
#   ^ source.just keyword.operator.recipe.end.just
>  -cat foo
#^^ source.just
#  ^ source.just keyword.operator.error-suppression.just
#   ^^^^^^^^ source.just
>  echo 'Done!'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^^ source.just string.quoted.single.just
#             ^ source.just string.quoted.single.just
>
>build target:
#^^^^^ source.just entity.name.function.just
#     ^ source.just
#      ^^^^^^ source.just variable.parameter.recipe.just
#            ^ source.just keyword.operator.recipe.end.just
>  @echo "Building {{target}}…"
#^^ source.just
#  ^ source.just keyword.operator.quiet.just
#   ^^^^^ source.just
#        ^ source.just string.quoted.double.just string.quoted.double.just
#         ^^^^^^^^^ source.just string.quoted.double.just
#                  ^^ source.just string.quoted.double.just string.interpolated.escaping.just string.interpolated.escape.just
#                    ^^^^^^ source.just string.quoted.double.just string.interpolated.escaping.just
#                          ^^ source.just string.quoted.double.just string.interpolated.escaping.just string.interpolated.escape.just
#                            ^ source.just string.quoted.double.just
#                             ^ source.just string.quoted.double.just
>
>push target: (build target)
#^^^^ source.just entity.name.function.just
#    ^ source.just
#     ^^^^^^ source.just variable.parameter.recipe.just
#           ^ source.just keyword.operator.recipe.end.just
#            ^ source.just
#             ^ source.just
#              ^^^^^ source.just entity.name.function.just
#                   ^^^^^^^ source.just
#                          ^ source.just
>  @echo 'Pushing {{target}}…'
#^^ source.just
#  ^ source.just keyword.operator.quiet.just
#   ^^^^^ source.just
#        ^ source.just string.quoted.single.just string.quoted.single.just
#         ^^^^^^^^ source.just string.quoted.single.just
#                 ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                   ^^^^^^ source.just string.quoted.single.just string.interpolated.escaping.just
#                         ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                           ^ source.just string.quoted.single.just
#                            ^ source.just string.quoted.single.just
>
># Extra
#^^^^^^^ source.just comment.line.number-sign.just
>
>test: lint && (build ("a" + ("b" / "c"))) && test
#^^^^ source.just entity.name.function.just
#    ^ source.just keyword.operator.recipe.end.just
#     ^ source.just
#      ^^^^ source.just entity.name.function.just
#          ^ source.just
#           ^^ source.just keyword.operator.and.just
#             ^ source.just
#              ^ source.just
#               ^^^^^ source.just entity.name.function.just
#                    ^^ source.just
#                      ^ source.just string.quoted.double.just string.quoted.double.just
#                       ^ source.just string.quoted.double.just
#                        ^ source.just string.quoted.double.just
#                         ^ source.just
#                          ^ source.just keyword.operator.concat.just
#                           ^^ source.just
#                             ^ source.just string.quoted.double.just string.quoted.double.just
#                              ^ source.just string.quoted.double.just
#                               ^ source.just string.quoted.double.just
#                                ^ source.just
#                                 ^ source.just keyword.operator.path-join.just
#                                  ^ source.just
#                                   ^ source.just string.quoted.double.just string.quoted.double.just
#                                    ^ source.just string.quoted.double.just
#                                     ^ source.just string.quoted.double.just
#                                      ^ source.just
#                                       ^ source.just
#                                        ^^ source.just
#                                          ^^ source.just keyword.operator.and.just
#                                            ^ source.just
#                                             ^^^^ source.just entity.name.function.just
>  @echo "test"
#^^ source.just
#  ^ source.just keyword.operator.quiet.just
#   ^^^^^ source.just
#        ^ source.just string.quoted.double.just string.quoted.double.just
#         ^^^^ source.just string.quoted.double.just
#             ^ source.just string.quoted.double.just
>
>default := 'all'
#^^^^^^^ source.just variable.other.just
#       ^ source.just
#        ^^ source.just keyword.operator.assignment.just
#          ^ source.just
#           ^ source.just string.quoted.single.just string.quoted.single.just
#            ^^^ source.just string.quoted.single.just
#               ^ source.just string.quoted.single.just
>
>test target tests=default:
#^^^^ source.just entity.name.function.just
#    ^ source.just
#     ^^^^^^ source.just variable.parameter.recipe.just
#           ^ source.just
#            ^^^^^ source.just variable.parameter.recipe.just
#                 ^ source.just keyword.operator.default.just
#                  ^^^^^^^ source.just
#                         ^ source.just keyword.operator.recipe.end.just
>  @echo 'Testing {{target}}:{{tests}}…'
#^^ source.just
#  ^ source.just keyword.operator.quiet.just
#   ^^^^^ source.just
#        ^ source.just string.quoted.single.just string.quoted.single.just
#         ^^^^^^^^ source.just string.quoted.single.just
#                 ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                   ^^^^^^ source.just string.quoted.single.just string.interpolated.escaping.just
#                         ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                           ^ source.just string.quoted.single.just
#                            ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                              ^^^^^ source.just string.quoted.single.just string.interpolated.escaping.just
#                                   ^^ source.just string.quoted.single.just string.interpolated.escaping.just string.interpolated.escape.just
#                                     ^ source.just string.quoted.single.just
#                                      ^ source.just string.quoted.single.just
>  ./test --tests {{tests}} {{target}}
#^^^ source.just
#   ^ source.just keyword.operator.path-join.just
#    ^^^^^^^^^^^^^ source.just
#                 ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                   ^^^^^ source.just string.interpolated.escaping.just
#                        ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                          ^ source.just
#                           ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                             ^^^^^^ source.just string.interpolated.escaping.just
#                                   ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
>test triple=(arch + "-unknown-unknown") input=(arch / "input.dat"):
#^^^^ source.just entity.name.function.just
#    ^ source.just
#     ^^^^^^ source.just variable.parameter.recipe.just
#           ^ source.just keyword.operator.default.just
#            ^ source.just
#             ^^^^ source.just support.function.builtin.just
#                 ^ source.just
#                  ^ source.just keyword.operator.concat.just
#                   ^ source.just
#                    ^ source.just string.quoted.double.just string.quoted.double.just
#                     ^^^^^^^^^^^^^^^^ source.just string.quoted.double.just
#                                     ^ source.just string.quoted.double.just
#                                      ^ source.just
#                                       ^ source.just
#                                        ^^^^^ source.just variable.parameter.recipe.just
#                                             ^ source.just keyword.operator.default.just
#                                              ^ source.just
#                                               ^^^^ source.just support.function.builtin.just
#                                                   ^ source.just
#                                                    ^ source.just keyword.operator.path-join.just
#                                                     ^ source.just
#                                                      ^ source.just string.quoted.double.just string.quoted.double.just
#                                                       ^^^^^^^^^ source.just string.quoted.double.just
#                                                                ^ source.just string.quoted.double.just
#                                                                 ^ source.just
#                                                                  ^ source.just keyword.operator.recipe.end.just
>  ./test {{triple}}
#^^^ source.just
#   ^ source.just keyword.operator.path-join.just
#    ^^^^^ source.just
#         ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#           ^^^^^^ source.just string.interpolated.escaping.just
#                 ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
># Extra case
#^^^^^^^^^^^^ source.just comment.line.number-sign.just
>test nested=("a" + ("b" + "c")):
#^^^^ source.just entity.name.function.just
#    ^ source.just
#     ^^^^^^ source.just variable.parameter.recipe.just
#           ^ source.just keyword.operator.default.just
#            ^ source.just
#             ^ source.just string.quoted.double.just string.quoted.double.just
#              ^ source.just string.quoted.double.just
#               ^ source.just string.quoted.double.just
#                ^ source.just
#                 ^ source.just keyword.operator.concat.just
#                  ^^ source.just
#                    ^ source.just string.quoted.double.just string.quoted.double.just
#                     ^ source.just string.quoted.double.just
#                      ^ source.just string.quoted.double.just
#                       ^ source.just
#                        ^ source.just keyword.operator.concat.just
#                         ^ source.just
#                          ^ source.just string.quoted.double.just string.quoted.double.just
#                           ^ source.just string.quoted.double.just
#                            ^ source.just string.quoted.double.just
#                             ^ source.just
#                              ^ source.just
#                               ^ source.just keyword.operator.recipe.end.just
>  ./test {{nested}}
#^^^ source.just
#   ^ source.just keyword.operator.path-join.just
#    ^^^^^ source.just
#         ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#           ^^^^^^ source.just string.interpolated.escaping.just
#                 ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>  echo (("a") + ("b" + "c"))
#^^^^^^^ source.just
#       ^ source.just
#        ^ source.just
#         ^ source.just string.quoted.double.just string.quoted.double.just
#          ^ source.just string.quoted.double.just
#           ^ source.just string.quoted.double.just
#            ^ source.just
#             ^ source.just
#              ^ source.just keyword.operator.concat.just
#               ^ source.just
#                ^ source.just
#                 ^ source.just string.quoted.double.just string.quoted.double.just
#                  ^ source.just string.quoted.double.just
#                   ^ source.just string.quoted.double.just
#                    ^ source.just
#                     ^ source.just keyword.operator.concat.just
#                      ^ source.just
#                       ^ source.just string.quoted.double.just string.quoted.double.just
#                        ^ source.just string.quoted.double.just
#                         ^ source.just string.quoted.double.just
#                          ^ source.just
#                           ^^ source.just
>
>commit MESSAGE *FLAGS:
#^^^^^^ source.just entity.name.function.just
#      ^ source.just
#       ^^^^^^^ source.just variable.parameter.recipe.just
#              ^ source.just
#               ^ source.just keyword.other.recipe.variadic.just
#                ^^^^^ source.just variable.parameter.recipe.just
#                     ^ source.just keyword.operator.recipe.end.just
>  git commit {{FLAGS}} -m "{{MESSAGE}}"
#^^^^^^^^^^^^^ source.just
#             ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#               ^^^^^ source.just string.interpolated.escaping.just
#                    ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#                      ^^^^ source.just
#                          ^ source.just string.quoted.double.just string.quoted.double.just
#                           ^^ source.just string.quoted.double.just string.interpolated.escaping.just string.interpolated.escape.just
#                             ^^^^^^^ source.just string.quoted.double.just string.interpolated.escaping.just
#                                    ^^ source.just string.quoted.double.just string.interpolated.escaping.just string.interpolated.escape.just
#                                      ^ source.just string.quoted.double.just
>
>test +FLAGS='-q':
#^^^^ source.just entity.name.function.just
#    ^ source.just
#     ^ source.just keyword.other.recipe.variadic.just
#      ^^^^^ source.just variable.parameter.recipe.just
#           ^ source.just keyword.operator.default.just
#            ^ source.just string.quoted.single.just string.quoted.single.just
#             ^^ source.just string.quoted.single.just
#               ^ source.just string.quoted.single.just
#                ^ source.just keyword.operator.recipe.end.just
>  cargo test {{FLAGS}}
#^^^^^^^^^^^^^ source.just
#             ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
#               ^^^^^ source.just string.interpolated.escaping.just
#                    ^^ source.just string.interpolated.escaping.just string.interpolated.escape.just
>
>foo $bar:
#^^^ source.just entity.name.function.just
#   ^ source.just
#    ^ source.just keyword.other.recipe.variadic.just
#     ^^^ source.just variable.parameter.recipe.just
#        ^ source.just keyword.operator.recipe.end.just
>  echo $bar
#^^^^^^^^^^^^ source.just
>
>a:
#^ source.just entity.name.function.just
# ^ source.just keyword.operator.recipe.end.just
>  echo 'A!'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^ source.just string.quoted.single.just
#          ^ source.just string.quoted.single.just
>
>b: a && c d
#^ source.just entity.name.function.just
# ^ source.just keyword.operator.recipe.end.just
#  ^ source.just
#   ^ source.just entity.name.function.just
#    ^ source.just
#     ^^ source.just keyword.operator.and.just
#       ^ source.just
#        ^ source.just entity.name.function.just
#         ^ source.just
#          ^ source.just entity.name.function.just
>  echo 'B!'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^ source.just string.quoted.single.just
#          ^ source.just string.quoted.single.just
>
>c:
#^ source.just entity.name.function.just
# ^ source.just keyword.operator.recipe.end.just
>  echo 'C!'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^ source.just string.quoted.single.just
#          ^ source.just string.quoted.single.just
>
>d:
#^ source.just entity.name.function.just
# ^ source.just keyword.operator.recipe.end.just
>  echo 'D!'
#^^^^^^^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^ source.just string.quoted.single.just
#          ^ source.just string.quoted.single.just
>
>@quiet:
#^ source.just keyword.other.recipe.prefix.just
# ^^^^^ source.just entity.name.function.just
#      ^ source.just keyword.operator.recipe.end.just
>  echo hello
#^^^^^^^^^^^^^ source.just
>  echo goodbye
#^^^^^^^^^^^^^^^ source.just
>  @# all done!
#^^ source.just
#  ^ source.just keyword.operator.quiet.just
#   ^^^^^^^^^^^ source.just comment.line.number-sign.just
>
>
># Imports
#^^^^^^^^^ source.just comment.line.number-sign.just
>
>import 'foo/bar.just'
#^^^^^^ source.just keyword.other.reserved.just
#      ^ source.just
#       ^ source.just string.quoted.single.just string.quoted.single.just
#        ^^^^^^^^^^^^ source.just string.quoted.single.just
#                    ^ source.just string.quoted.single.just
>
>mod bar
#^^^ source.just keyword.other.reserved.just
#   ^ source.just
#    ^^^^ source.just
>